src/manager.js

```diff
@@ -106,24 +109,76 @@ AdManager.prototype.initGoogleTag = function() {
  *
  * @returns undefined
 */
+
+/** runs auctions on individual adunits
+ * might want to batch these like prebid
+*/
 AdManager.prototype.fetchAmazonBids = function(elementId, gptSizes, slotName) {
-	var adUnitPath = this.getAdUnitCode(),
-	slotUnit = adUnitPath + '_' + slotName;
-	window.apstag.fetchBids({
-		slots: [{
-			slotID: elementId,
-			sizes: gptSizes,
-			slotName: slotUnit
-		}],
-		timeout: 1e3
-	}, callback = function (bids) {
-		// Your callback method, in this example it triggers the first DFP request for googletag's disableInitialLoad integration after bids have been set
-		window.headertag.cmd.push(function () {
-			window.apstag.setDisplayBids();
-		});
-	});
+  var adUnitPath = this.getAdUnitCode(),
+  slotUnit = adUnitPath + '_' + slotName;
+  window.apstag.fetchBids({
+    slots: [{
+      slotID: elementId,
+      sizes: gptSizes,
+      slotName: slotUnit
+    }],
+    timeout: 1e3
+  }, callback = (bids) => {
+    // Your callback method, in this example it triggers the first DFP request for googletag's disableInitialLoad integration after bids have been set
+    if (typeof window.headertag !== 'undefined') {
+      window.headertag.cmd.push(function() {
+          window.apstag.setDisplayBids();
+      });
+    } else {
+        window.apstag.setDisplayBids();
+    }
+  });
+};
+
+AdManager.prototype.addUnitToPrebid = function(adUnitConfig, activeSizes, slot) {
+  const adUnitPath = this.getAdUnitCode();
+  const { prebid } = adUnitConfig;
+
+  if(!prebid) return;
+
+  const code = slot.getSlotElementId();
+  clearTimeout(this.prebidAuctionCountdown);
+
+  const pbjsConfig = Object.assign({}, prebid);
+
+  Object.keys(pbjsConfig.mediaTypes).forEach(key => {
+    pbjsConfig.mediaTypes[key]['sizes'] = activeSizes;
+  });
+
+  this.prebidAuctionSlots = [...this.prebidAuctionSlots || [], slot];
+  this.pbjs.addAdUnits(Object.assign({code: code}, pbjsConfig));
+
+  this.prebidAuctionCountdown = setTimeout(() => {
```

Comment:
Can you use a named function for this, so we can wrap with unit tests?  Maybe already part of your plan.  If you're doing it to be able to use `this` values, you should still be able to bind it